### PR TITLE
Support `error` websocket channel

### PR DESCRIPTION
**Pull request by giannello**
This PR stems from jenkins-kubernetes not being able to properly deal with the return code when `exec`ing into a container. The kubernetes-client doesn't provide a separate pipe for the error channel exposed over the websocker, merging it with `stderr` instead.

This changeset exposes the `error` channel as a separate stream, and renames the others to keep a consistent naming strategy. The `resize` channel is deliberately ignored.

### CODE REVIEW

**Comment by giannello** [comment]
@carlossg  https://issues.jenkins-ci.org/browse/JENKINS-49762

**Comment by piyush1594** [comment]
ok to test

**Comment by piyush1594** [comment]
retest

**Comment by carlossg** [comment]
Thanks @giannello ! 
I gave it a try but I think it's still missing the methods in `PodOperationsImpl` in order to pass the stream where you would want errors to be written.

It can get confusing given that there's already 
* writingError
* readingError
* redirectingError

`Errorable` interface also needs to be changed or a new one created.

And is the header `Sec-WebSocket-Protocol: v4.channel.k8s.io` needed too? https://github.com/kubernetes-client/python/issues/297

**Comment by giannello** [comment]
@carlossg added you as a collaborator in case you want to submit patches. Will take a look at the classes you mentioned tomorrow.
Not sure about the header, `v4` should be the default if nothing else is specified - I don't really have an opinion and I'm around here just because I stumbled on that python client PR that mentioned the existence of a dedicated signaling channel, so please don't rely on my java skills.
Thank you for quickly jumping in, very appreciated.

**Comment by carlossg** [comment]
In the meantime and given how many name changes are involved I'd love to hear from the maintainers before implementing the next changes

**Comment by carlossg** [comment]
I have added the needed methods, and the implementation on https://github.com/jenkinsci/kubernetes-plugin/pull/300 to use them

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java [hunk 1]
```diff
@@ -232,18 +244,18 @@ public void onMessage(WebSocket webSocket, ByteString bytes) {
...
-                            err.write(byteString.toByteArray());
+                        if (stderr != null) {
+                            stderr.write(byteString.toByteArray());
                         }
                         break;
                     case 3:
-                        if (err != null) {
-                            err.write(byteString.toByteArray());
+                        if (error != null) {
+                            error.write(byteString.toByteArray());
```
Comment (carlossg) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java]: this is the interesting part, not sending error stream to stderr stream

**Comment by carlossg** [comment]
This is ready, tests in kubernetes-plugin are passing using the new methods, would love to hear about the API changes and method naming from a maintainer
https://github.com/jenkinsci/kubernetes-plugin/pull/300

@iocanel ?

**Comment by oscerd** [comment]
ping @iocanel

**Comment by calebdelnay** [comment]
I'm wondering if there is anything a random outsider could do to help push this forward?

We use Jenkins to build Docker images using workers allocated via the Kubernetes plugin. However due to https://github.com/jenkinsci/kubernetes-plugin/pull/300, which in turn depends on this issue, we're doing a manual build of the Kubernetes plugin with these fixes applied. It would be great to have these merged and included in upcoming releases.

**Comment by iocanel** [comment]
Don't know how I missed this.
Will have a look shortly.

**Comment by carlossg** [comment]
Ping @bparees

**Comment by bparees** [comment]
@piyush1594 @iocanel bump, we definitely want to help enable the kubernetes pipeline plugin.

**Comment by carlossg** [comment]
I wonder if it would be useful to avoid all the renaming for the sake of simplicity first so you can better see the actual changes needed

**Comment by giannello** [comment]
I can push a less intrusive change, will find some time this afternoon to find a better naming.

**Comment by carlossg** [comment]
any ETA for merging and releasing?

**Comment by iocanel** [comment]
@giannello @carlossg: Apologies for the long delay. Since my main concern was naming and its addressed I'll merge asap.

**Comment by carlossg** [comment]
@iocanel how long until it makes it into a release ?

**Comment by piyush1594** [comment]
Hey, @iocanel Thinking of releasing the client. We need to bump the minor version or it is not required for the time being

**Comment by piyush1594** [comment]
Hey @carlossg Kubernetes Client 3.2.0 released which have this feature.

**Pull request merged**